build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionBwcSetupPlugin.java

```diff
@@ -249,8 +250,14 @@ static void createBuildBwcTask(
                 @Override
                 public void execute(Task task) {
                     if (expectedOutputFile.exists() == false) {
+                        Path relativeOutputPath = project.getRootDir().toPath().relativize(expectedOutputFile.toPath());
                         throw new InvalidUserDataException(
-                            "Building " + bwcVersion.get() + " didn't generate expected artifact " + expectedOutputFile
+                            "Building "
```

Comment:
Maybe we can go further and explicitly suggest a merge?

```java
final String message = "Building %s didn't generate expected artifact [%s]. The working branch may be "
    + "out-of-date - try merging in the latest upstream changes to the branch.";
throw new InvalidUserDataException(message.formatted(bwcVersion.get(), relativeOutputPath));
```

Reply:
Yes, that is indeed better.